examples/model-score/plotly_streaming.py

```diff
@@ -0,0 +1,100 @@
+import functools
+import json
+
+import plotly.graph_objects as go
+from shinywidgets import render_widget
+
+from shiny import reactive
+
+
+# Return a hash of an arbitrary object, including nested dicts, lists, and numpy/pandas
+# data structures. Uses json.dumps() internally.
+def _hash_anything(obj):
+    return hash(json.dumps(obj, sort_keys=True, default=_to_json_repr))
+
+
+def _to_json_repr(obj):
+    # If has to_json(), use that; make sure it's callable
+    if hasattr(obj, "to_json") and callable(obj.to_json):
+        return json.loads(obj.to_json())
+    if hasattr(obj, "to_list") and callable(obj.to_list):
+        return obj.to_list()
+    if hasattr(obj, "tolist") and callable(obj.tolist):
+        return obj.tolist()
+    if hasattr(obj, "to_dict") and callable(obj.to_dict):
+        return obj.to_dict()
+    raise TypeError(f"Object of type {type(obj)} is not JSON serializable")
+
+
+def render_plotly_streaming(
```

Comment:
This is awesome, I wonder if we could do something similar with matplotlib.

Reply:
That would be super helpful, the matplotlib rendering time definitely limits its usefulness in realtime apps.

Reply:
I raised this issue https://github.com/rstudio/py-shiny/issues/653